doc/riscv-bfloat16-appx-rationale.adoc

```diff
@@ -0,0 +1,67 @@
+[appendix]
+[[BFloat16_appx_rationale]]
+= Extension Rationale
+
+== Format Rationale
+Various choices were made in the RISC-V BFloat16 format and behavior.
+Some of these choices are allowed by IEEE-754 and others are deviations
+from the standard
+
+=== Rounding Modes
+
+==== Round to odd
+Round to odd is not a '754 supported rounding mode. However, it avoids double
+rounding can occur when accumulating a result in a wider format and then
+converting the result to a narrower format before subsequent usage.
+
+==== Round to nearest - even
+Round to nearest, ties to even is the default '754 rounding format. It is unbiased
+and minimze rounding error.
+
+=== Subnormal Handling
+
+=== NaN handling
+
+=== Zeros and Infinities
+
+== Instruction Rationale
+
+This section contains various rationale, design notes and usage
+recommendations for the instructions in the BFloat16 extension.
+It also tries to record how the designs of instructions were
+derived, or where they were contributed from.
+
+=== Conversion Instructions
+
+
+The most common and important conversion instructions are between BFloat16 and FP32
+(Single Precision).
+
+We chose not to have direct conversion between BFloa16 and other formats as they
+can typcially be performed by a combination of instructions. 
+
+.Notes to software developers
+[NOTE,caption="SH"]
+====
+In some cases, for example convert from FP64 to BFloat16 there can be double rounding.
+It is up to software to elimiante such sources of error if this is important to the
+application.
+====
+
+=== FMA
+
+Fused multiply add.
+
+=== Dot Product

```

Comment:
Is the intention here to support dot product as a packed-SIMD style operation, or an application of FMA?

Reply:
Upon reading the rest of the spec, it seems like this is intended to be used in packed-SIMD style ops. The spec should probably also define other operations on multiple packed BF16 operands (at least a note on how to load/store them - probably using standard FLW/FLD?) and how this is intended to work in general; I assume it's useful to to think about this for all operations, since just using 16 bits of a 32 or 64 bit register seems a bit wasteful.

Reply:
Yes, the intention is for dot product operations, as that is what BF16 is usually used for. It was requested that we start with the base and then we can move on to operations. I anticipate that these operations will be most useful in Vector.